### PR TITLE
Add instance method to PyTorch frontend

### PR DESCRIPTION
**Pull request by beandkay**
Close https://github.com/unifyai/ivy/issues/23059
Close https://github.com/unifyai/ivy/issues/23071

### CODE REVIEW

**Comment by beandkay** [comment]
@satakshigarg can you check it?

**Comment by beandkay** [comment]
@satakshigarg @Dsantra92 anyone can help me check it?

**Comment by beandkay** [comment]
@satakshigarg hello?

**Comment by beandkay** [comment]
@satakshigarg @Dsantra92 hello?

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -100,9 +100,20 @@ def requires_grad(self):
...
+        if self.device == "cpu":
+            return -1
+        else:
+            return int(self.devices.split(':')[-1])
+
     # Setters #
     # --------#
 
+    @device.setter
+    def cuda(self):
```
Comment (Dsantra92) [inline ivy/functional/frontends/torch/tensor.py:114]: Can you conform it to the original torch method [here](https://pytorch.org/docs/stable/generated/torch.Tensor.cuda.html). We should able to mention the device at-least.

**Review by Dsantra92 — COMMENTED** [review]
The code looks good to me, was the code formatted by the linter?

File: ivy/functional/frontends/torch/tensor.py (hunk 1 above)
Comment (beandkay) [reply ivy/functional/frontends/torch/tensor.py]: @Dsantra92 can you check it again

Comment (beandkay) [reply ivy/functional/frontends/torch/tensor.py]: @Dsantra92 hello?

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -105,14 +105,14 @@ def get_device(self):
...
+            return int(self.device.split(':')[-1])
 
     # Setters #
     # --------#
 
     @device.setter
-    def cuda(self):
-        self.device = "gpu:0"
+    def cuda(self, device=None):
+        return self.device = device
```
Comment (Dsantra92) [inline ivy/functional/frontends/torch/tensor.py:115]: This perhpas would make more sense here:
```python
self.device = device
return self
```

**Comment by Dsantra92** [comment]
Sorry for the late reply. If you are waiting for a review, feel free to ping on discord. Anyway the PR looks good to me, a small change it would good to merge.

**Comment by beandkay** [comment]
@Dsantra92 thank you, i fixed the code.

**Comment by beandkay** [comment]
@Dsantra92 hello? i did ping on discord.

**Comment by Dsantra92** [comment]
Hey @beandkay, sorry for the late update. This PR introduces new failures, as you can check  [here](https://github.com/unifyai/ivy/actions/runs/6192435343/job/16812943559?pr=23060)under New Failures Introduced. We need to fix the issues before the PR is merged.

**Comment by beandkay** [comment]
@Dsantra92 I don't know why those test cases are failed, I init the dtype to 'valid' but it keeps providing a 'uint' type to torch framework.

**Comment by beandkay** [comment]
@Dsantra92 hello. please check it

**Comment by beandkay** [comment]
@Dsantra92 hello?

**Comment by beandkay** [comment]
@Dsantra92

**Comment by beandkay** [comment]
@Dsantra92 please follow this please

**Comment by beandkay** [comment]
@Dsantra92 hello?

**Comment by beandkay** [comment]
@Dsantra92 @YushaArif99 pls check it ....

**Comment by beandkay** [comment]
@Dsantra92  hello, are you still follow this

**Pull request merged**